src/popupmenu.c

```diff
@@ -774,33 +805,40 @@ pum_redraw(void)
 			width += w;
 		}
 
-	    if (round > 0)
-		n = pum_kind_width + 1;
+	    if (j > 0)
+		n = items_width_array[order[1]] + (last_isabbr ? 0 : 1);
 	    else
-		n = 1;
+		n = order[j] == CPT_ABBR ? 1 : 0;
+
+	    if (j + 1 < 3)
+	    {
+		if (pum_order_item_isabbr(order, j + 1))
+		    next_isempty = pum_array[idx].pum_text == NULL;
+		else
+		    next_isempty = pum_get_item(idx, order[j + 1]) == NULL;
```

Comment:
Same here, why not use `pum_get_item()` directly?

Reply:
function is abstracted later. So I forgot some adjustments.